dodo.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+import subprocess
+from pathlib import Path
+from subprocess import PIPE
+
+import doit
+from doit import task_params
+from doit.tools import CmdAction
+from tests.images_hierarchy import get_test_dirs
+
+# global doit config
+DOIT_CONFIG = {"verbosity": 2, "default_tasks": ["build_docs"]}
+
+
+# dependencies: input to the task execution -> keeps tracks of the state of file dependencies and saves the signature
+# of them every time the tasks are run so if there are no modifications to the files the execution of the task is skipped
+# (indicated by -- after running doit)
+# target: output produced by the task execution
+
+# -----------------------------------------------------------------------------
+# Documentation and wiki tasks
+# -----------------------------------------------------------------------------
+
+
+def task_build_docs():
+    """Build Sphinx documentation 📝
+    setting uptodate to False will force the task to run every time
+    """
+
+    return dict(
+        file_dep=[*P.DOCS_MD, *P.DOCS_RST, *P.DOCS_PY],
+        actions=[U.do("sphinx-build", "-W", P.DOCS, "docs/_build/html")],
+        targets=[P.DOCS_TARGET],
+        uptodate=[False],
+    )
+
+
+# https://pydoit.org/task-creation.html#delayed-task-creation
+@doit.create_after(executed="build_docs")
+def task_docs_check_links():
+    """Checks for any broken links in the Sphinx documentation 🔗
+    only created after the docs are built"""
+    return dict(
+        file_dep=[*P.DOCS.rglob("_build/*.html")],
```

Comment:
As you put the html files in the html subfolder, this rglob will probably give empty output.

Reply:
Now, I've put the files in the right dir, so this should be fine.